c_emulator/rv_cfg_func.c

```diff
@@ -0,0 +1,452 @@
+//    printf("%s, %d: zxlen_val: %ld\n", __FILE__, __LINE__, zxlen_val);
+//    if (zxlen_val == 32) 
+//        {
+//        if (!rv_cfg_c_ext_enable(RV_ISA, "RV32"))
+//            {
+//            fprintf(stderr, "%s, %d: incompatable settings for zxlen_val and RV_ISA. both must be 32.", 
+//                    __FILE__, __LINE__);
+//            exit(1);
+//            }
+//        RV32ISA = RV_ISA;
+//        }
+//    else if (zxlen_val == 64) 
+//        {
+//        if (!rv_cfg_c_ext_enable(RV_ISA, "RV64"))
+//            {
+//            fprintf(stderr, "%s, %d: incompatable settings for zxlen_val and RV_ISA. both must be 64.", 
+//                    __FILE__, __LINE__);
+//            exit(1);
+//            }
+//        RV64ISA = RV_ISA;
+//        }
+//    else if (zxlen_val == 128) 
+//        {
+//        if (!rv_cfg_c_ext_enable(RV_ISA, "RV128"))
+//            {
+//            fprintf(stderr, "%s, %d: incompatable settings for zxlen_val and RV_ISA. both must be 128.", 
+//                    __FILE__, __LINE__);
+//            exit(1);
+//            }
+//        RV128ISA = RV_ISA;
+//        }
+//    else
+//        {
+//        fprintf(stderr, "%s, %d: invalid setting for zxlen_val: %d.", 
+//                __FILE__, __LINE__, zxlen_val);
+//        }
+
+    // ====================================
+    // Check extension settings for mutual exclusivity
+    // TODO:  should I introduce 'assert' as an error reporting mechanism
+    //      instead of this ad hoc mechanism?
+    if ((rv_enable_fdext == true) && (rv_enable_zfinx == true))
+        {
+        fprintf(stderr, "%s, %d: incompatable settings for [fd]ext and Zfinx. both cannot be enabled.", 
+                __FILE__, __LINE__);
+        exit(1);
+        }
+
+    }
+
+// ============================================================================
+bool
+rv_cfg_c_path_exists(char * path)
+    {
+    struct fy_path_parse_cfg    cfg;
+    struct fy_path_parser       *fypp;
+    bool                        ret;
+
+    cfg.flags = FYPPCF_QUIET;
+    cfg.userdata = NULL;
+    cfg.diag = NULL;
+
+    fypp = fy_path_parser_create(&cfg);
+    ret = (fy_path_parse_expr_from_string(fypp, path, -1) == NULL) ? false : true;
+    fy_path_parser_destroy(fypp);
+
+    return(ret);
+    }
+
+// ============================================================================
+bool
+rv_cfg_c_bool(char * key_str)
+    {
+    char    *s;
+
+    s = rv_cfg_c_string(key_str);
+
+    if ( strcmp("true", s) == 0)
+        {
+        return(true);
+        }
+    else if ( strcmp("false", s) == 0 )
+        {
+        return(false);
+        }
+    else
+        {
+        fprintf(stderr, "%s, %d: internal error. '%s' does not appear to be a bool.\n", 
+                __FILE__, __LINE__, key_str);
+        exit(1);
+        }
+    }
+
+// ============================================================================
+int
+rv_cfg_c_ext_enable(char * isa_str, char * ext_pattern)
```

Comment:
This function only receives string constants, so there should be no need for pcre2? strstr() should be sufficient?